vadl/main/vadl/lcb/template/lib/Target/AsmParser/EmitAsmParserCppFilePass.java

```diff
@@ -166,23 +194,33 @@ private List<TableGenOperand> createOperands(TableGenInstruction instruction) {
     for (var input : instruction.getInOperands()) {
       var casted = (GcbDefaultInstructionOperand) input;
       if (input instanceof TableGenInstructionImmediateOperand immediateOperand) {
+
         var operand = new TableGenOperand(immediateOperand.name(),
             indexOffset,
             true,
-            immediateOperand.immediateOperand().predicateMethod().lower()
+            immediateOperand.immediateOperand().predicateMethod().lower(),
+            true,
+            immediateOperand.formatFields().get(0).simpleName(),
```

Comment:
major: Regardless what the final implementation will look like, this is a silent foot gun when there is an encoding function with multiple field access functions. I would prefer it, if we throw a `DiagnosticError` when an operand has multiple format fields to spot the error easier and quicker.